File: leetcode_solutions/0134-gas-station/solution.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int totalUsed = 0;
        int currentUsed = 0;
        int iStart = 0;
        for (int i = 0; i < gas.size(); i++){
            
            currentUsed += gas[i] - cost[i];
            totalUsed += gas[i] - cost[i];
        
            if (currentUsed < 0){
                iStart = i + 1;
                currentUsed = 0;
            }
        }
        return totalUsed >= 0 ? iStart : -1;
    }
};

File: leetcode_solutions/0134-gas-station/solution.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int total_tank = 0;
        int curr_tank = 0;
        int starting_station = 0;
        for (int i = 0; i < gas.length; i++){
            total_tank += gas[i] - cost[i];
            curr_tank += gas[i] - cost[i];

            if (curr_tank < 0){
                starting_station = i + 1;
                curr_tank = 0;
            }
        }

        return total_tank >= 0 ? starting_station : -1;
    }
}
